### PR TITLE
Chore/reorg scss

### PR DESCRIPTION
**Pull request by itsamoreh**
NOTE: This is a follow up PR for #27 that fixes some issues after rebasing on `main`.

# Closes

N/A

## Link to test

N/A

## Description

* Reorganize scss files and directories
  * Make directories more literal (e.g. `template-parts` to style WordPress Template Parts)
  * Use `_index.scss` so we can get rid of the double import (`@import './base'` instead of `@import './base/base'`)
  * Remove underscore from scss files other than index, so that index can be at the top of the dir
    * The underscores are not required anyway, this keeps things nice and organized visually
  * Remove the `components` directory and move form styles to `base`
    * In the spirit of literal directory names, since there is nothing called "components" in WP, this PR removes this dir and moves the form styles to `base` since they style bare html elements like `input` and `select`

## Added to documentation?

- [x] 📜 README.md

## Testing Instructions

1. Pull this branch, build the theme
2. Verify that styles are building correctly

-----

## Reviewer's Testing Checklist

As a reviewer, please verify that the relevant testing criteria are fulfilled and confirmed before approving this Pull Request.

- [ ] **Visual Regression Testing:** Ensure that existing functionality is not negatively impacted by the changes.
- [ ] **Cross-Browser Compatibility:** Test on major browsers (Chrome, Firefox, Safari) to ensure compatibility.
- [ ] **Mobile Responsiveness:** Confirm that the changes are responsive and functional on various mobile devices.
- [ ] **Theme Compatibility:** Ensure that the changes do not adversely affect the site's theme and styling.
- [ ] **Linting:** Check that the code passes all linting checks (PHPCS, ESLint, SassLint). Check if PR passes code quality check.
- [ ] **Accessibility Testing:** Validate that the changes comply with accessibility standards. Run `npm run a11y`.
- [ ] **Security Best Practices:** Ensure that the code follows WordPress security best practices. Check if PR passes security check.
- [ ] **Documentation:** Ensure that any new features or changes are appropriately documented in the README.md or Confluence.
- [ ] **Post-Deployment Tasks:** Check if there are any tasks that need to be performed after deployment.

### CODE REVIEW

**Comment by khleomix** [comment]
@itsamoreh thanks! This was in my to-do. I'll also need to rename from assets to src since that is the default folder, that way, we don't have to use the dir flag for wp-scripts. I'll do that since that might require quiet a few changes in webpack config. Will review once I'm back on Monday.

File: .gitignore
Comment (khleomix) [inline .gitignore:1]: @itsamoreh the top level blocks folder is part of the build process so this still needs to be ignored.

File: .stylelintignore
Comment (khleomix) [inline .stylelintignore:1]: @itsamoreh same here for the blocks folder.

**Review by khleomix — COMMENTED** [review]
@itsamoreh just a couple things with the generated blocks folder

File: .gitignore
Comment (itsamoreh) [reply .gitignore:1]: Added it back, targeting the top level `/blocks` dir only - [5d91214](https://github.com/WebDevStudios/wds-bt/pull/28/commits/5d912141bde9a9ee2a31f80e3ed3f0a4a4994444)

File: .stylelintignore
Comment (itsamoreh) [reply .stylelintignore:1]: Added it back, targeting the top level `/blocks` dir only - [5d91214](https://github.com/WebDevStudios/wds-bt/pull/28/commits/5d912141bde9a9ee2a31f80e3ed3f0a4a4994444)

**Review by khleomix — APPROVED** [review]
Thanks!

**Pull request merged**